assets/blocks/course-outline/outline-block/existing-lessons-modal/lessons.js

```diff
@@ -0,0 +1,209 @@
+				( lesson ) =>
+					! lesson.meta._lesson_course ||
+					lesson.meta._lesson_course !== courseId
+			);
+
+			const courseIds = foundLessons
+				? foundLessons
+						.filter( ( lesson ) => lesson.meta._lesson_course )
+						.map( ( lesson ) => lesson.meta._lesson_course )
+				: [];
+
+			const courses = select( 'core' ).getEntityRecords(
+				'postType',
+				'course',
+				{
+					per_page: 100,
+					include: courseIds,
+				}
+			);
+
+			// Add course field to lessons.
+			if ( foundLessons && courses ) {
+				foundLessons = foundLessons.map( ( lesson ) => {
+					const lessonCourseId = lesson.meta._lesson_course;
+					if ( ! courseId ) {
+						return lesson;
+					}
+
+					let course = courses.find(
+						( c ) => c.id === lessonCourseId
+					);
+
+					if ( ! course ) {
+						course = {
+							id: undefined,
+							title: {
+								raw: __( 'Course not assigned', 'sensei-lms' ),
+							},
+						};
+					}
+
+					return {
+						...lesson,
+						course,
+					};
+				} );
+			}
+
+			return foundLessons;
+		},
+		[ filters ]
+	);
+
+	if ( ! lessons ) {
+		return (
+			<div className="sensei-lms-quiz-block__existing-lessons-modal__existing-lessons sensei-lms-quiz-block__existing-lessons-modal__existing-lessons--loading">
+				<Spinner />
+			</div>
+		);
+	}
+
+	// Filter out already added lessons.
+	lessons = lessons.filter(
+		( lesson ) => ! addedLessonIds.includes( lesson.id )
+	);
+
+	const allChecked =
+		lessons.length > 0 &&
+		lessons.every( ( lesson ) => selectedLessonIds.includes( lesson.id ) );
+
+	const toggleAllHandler = ( checked ) => {
+		const lessonIds = lessons.map( ( lesson ) => lesson.id );
+
+		setSelectedLessonIds( ( prev ) =>
+			checked
+				? uniq( [ ...prev, ...lessonIds ] )
+				: prev.filter( ( lesson ) => ! lessonIds.includes( lesson ) )
+		);
+	};
+
+	const toggleLesson = ( lessonId ) => ( checked ) => {
+		if ( checked ) {
+			setSelectedLessonIds( ( prev ) => [ ...prev, lessonId ] );
+		} else {
+			setSelectedLessonIds( ( prev ) =>
+				prev.filter( ( id ) => id !== lessonId )
+			);
+		}
+	};
+
+	const lessonsMap = ( lesson ) => {
+		const course =
+			lesson.course?.title.raw || __( 'Loading…', 'sensei-lms' );
+		const courseNotFoundClass =
+			lesson.course?.id === undefined
+				? 'sensei-lms-quiz-block__existing-lessons-modal__course-title--not-found'
+				: '';
+		const lessonId = lesson.id;
+		const title = lesson.title.raw;
+
+		return (
+			<tr key={ lessonId }>
+				<td>
+					<CheckboxControl
+						id={ `existing-lesson-${ lessonId }` }
+						title={ title }
+						checked={ selectedLessonIds.includes( lessonId ) }
+						onChange={ toggleLesson( lessonId ) }
+					/>
+				</td>
+				<td className="sensei-lms-quiz-block__existing-lessons-modal__lesson-title">
+					<label
+						htmlFor={ `existing-lesson-${ lessonId }` }
+						title={ title }
+					>
+						{ title }
+					</label>
+				</td>
+				<td className={ courseNotFoundClass }>{ course }</td>
```

Comment:
Yep, sounds good. 
Do you know if we have something like that in Sensei or a best practice? 
(I'm also looking on my own; I asked just in case you already know.)

Reply:
I searched for the `info` icon, and I found this: https://github.com/Automattic/sensei/pull/4001

Probably not exactly what you want here, but maybe some parts could be useful as reference?

Reply:
No need for that anymore :)